sphinx/tutorial/rose/furthertopics/trigger.rst

```diff
@@ -0,0 +1,241 @@
+pretty irrelevant if we're ordering a ``'Veggie Supreme'``.
+
+
+Adding Triggers
+---------------
+
+Let's add some trigger information.
+
+In the ``rose-meta.conf`` file, under ``[namelist:pizza_order=pizza_type]``,
+add:
+
+.. code-block:: cylc
+
+   trigger=namelist:pizza_order=extra_chicken: 'BBQ Chicken';
+           namelist:pizza_order=pepperoni_multiple: 'Pepperoni', 'BBQ Chicken';
+
+This states which values of ``pizza_type`` are relevant for which
+settings. This means that ``extra_chicken`` is only relevant when
+``pizza_type`` is ``'BBQ Chicken'`` - otherwise, it should be in
+an ignored state. ``pepperoni_multiple`` is relevant for more than
+one value of ``pizza_type``.
+
+We should also make sure we don't order over our budget, especially
+by splashing out on truffles. Add the following to ``[env=BUDGET]``:
+
+.. code-block:: cylc
+
+   trigger=namelist:pizza_order=truffle: this > 25;
+           namelist:side_order: this >= 10;
+
+.. TODO - link the logical syntax reference page.
+
+What we've done here is use a small subset of the Rose configuration
+metadata logical syntax to specify a range of allowed values (the
+``this > 25`` part). Here, ``this`` is a placeholder for the value of
+``env=BUDGET``; the expression syntax is essentially Pythonic.
+
+We've also specified a section ``namelist:side_order`` in the trigger,
+which is perfectly valid - this means that the whole section and its
+options will be ignored when the value of ``env=BUDGET`` is below 10.
+The truffle option will be ignored unless ``env=BUDGET`` is more than
+25.
+
+
+Fixing Trigger Errors
+---------------------
+
+If we load the config editor (or reload the metadata) again, we
+should get some trigger errors. These essentially say that some of
+our settings are in the wrong state now - in our case, they should
+be ``trigger-ignored``.
+
+You can fix them on the command line by running ``rose macro --fix``
+or ``rose macro -F`` in the app directory (one level up from the
+meta directory) - this is what you would do if you were working with
+a text editor and made changes to values.
+
+Similarly, you can run "Autofix" in the config editor. You can do this
+in three ways:
+
+* By clicking the :menuselection:`Metadata --> Autofix all configurations`
+  menu.
+* Using the :guilabel:`Auto-fix` toolbar button.
+* Or via the right click menu for the root page in the left hand tree panel,
+  in this case ``pizza_order``.
+
+Run "Autofix" in one of the above ways.
+
+Results
+^^^^^^^
+
+If you accept the changes, the state of these settings will be corrected
+- if you go to the page, you'll see that they've vanished! They're
+actually just commented out, and viewable via the menu
+:menuselection:`View --> View All Ignored Variables`.
+
+Try altering the values of ``namelist:pizza_order=pizza_type`` and
+``env=BUDGET`` - with :menuselection:`View --> View All Ignored Variables`
+on and off. This should enable and ``trigger-ignore`` different settings.
```

Comment:
Is ``trigger-ignore`` a command as well as ``trigger-ignored`` with the 'd' on the end? If not I would remove the monospace highlighting in these such cases.